components/support/rc_crypto/src/ece_crypto.rs

```diff
@@ -159,102 +163,27 @@ impl CryptoImpl {
-    }
-
-    fn compute_ecdh_secret(
-        remote: &Self::RemotePublicKey,
-        local: &Self::LocalKeyPair,
-    ) -> Result<Vec<u8>> {
-        convert(Self::_compute_ecdh_secret(remote, local))
-    }
-
-    fn hkdf_sha256(salt: &[u8], secret: &[u8], info: &[u8], len: usize) -> Result<Vec<u8>> {
-        convert(Self::_hkdf_sha256(salt, secret, info, len))
-    }
-
-    fn aes_gcm_128_encrypt(key: &[u8], iv: &[u8], data: &[u8], tag_len: usize) -> Result<Vec<u8>> {
-        convert(Self::_aes_gcm_128_encrypt(key, iv, data, tag_len))
-    }
-
-    fn aes_gcm_128_decrypt(key: &[u8], iv: &[u8], data: &[u8], tag: &[u8]) -> Result<Vec<u8>> {
-        convert(Self::_aes_gcm_128_decrypt(key, iv, data, tag))
-    }
-
-    fn random(dest: &mut [u8]) -> ece::Result<()> {
-        convert(Self::_random(dest))
-    }
-}
-
-pub type Aes128GcmEceWebPushImpl = Aes128GcmEceWebPush<CryptoImpl>;
-pub type AesGcmEceWebPushImpl = AesGcmEceWebPush<CryptoImpl>;
-
-/// Generate a local ECE key pair and auth nonce.
-pub fn generate_keypair_and_auth_secret(
-) -> Result<(LocalKeyPairImpl, [u8; ECE_WEBPUSH_AUTH_SECRET_LENGTH])> {
-    let local_key_pair = LocalKeyPairImpl::generate_random()?;
-    let mut auth_secret = [0u8; ECE_WEBPUSH_AUTH_SECRET_LENGTH];
-    CryptoImpl::random(&mut auth_secret)?;
-    Ok((local_key_pair, auth_secret))
-}
-
-/// Encrypt a block using default AES128GCM encoding.
-///
-/// param remote_pub &[u8] - The remote public key
-/// param remote_auth &u8 - The remote authorization token
-/// param salt &[u8] - The locally generated random salt
-/// param data &[u8] - The data to encrypt
-///
-pub fn encrypt(remote_pub: &[u8], remote_auth: &[u8], salt: &[u8], data: &[u8]) -> Result<Vec<u8>> {
-    let remote_key = RemotePublicKeyImpl::from_raw(remote_pub)?;
-    let local_key = CryptoImpl::generate_ephemeral_keypair()?;
-    // Pick a random size to pad to.
-    // It's a sampled random, endianness doesn't really matter here.
-    // XXX TODO: could we pick too much padding here, and fail with EncryptPadding error?
-    let mut padr = [0u8; 2];
-    CryptoImpl::random(&mut padr)?;
-    let pad = ((usize::from(padr[0]) + (usize::from(padr[1]) << 8)) % 4095) + 1;
-    let params = WebPushParams::new(4096, pad, Vec::from(salt));
-    Ok(Aes128GcmEceWebPushImpl::encrypt_with_keys(
-        &local_key,
-        &remote_key,
-        &remote_auth,
-        data,
-        params,
-    )?)
-}
-
-/// Decrypt a block using default AES128GCM encoding.
-///
-/// param components &str - The locally generated private key components.
-/// param auth &str - The locally generated auth token (this value was shared with the encryptor)
-/// param data &[u8] - The encrypted data block
-///
-pub fn decrypt(components: &EcKeyComponents, auth: &[u8], data: &[u8]) -> Result<Vec<u8>> {
-    let priv_key = LocalKeyPairImpl::from_raw_components(components).unwrap();
-    Aes128GcmEceWebPushImpl::decrypt(&priv_key, &auth, data)
+// Note: this doesn't initialize NSS!
```

Comment:
As a reader, this comment is obviously important, but it's not clear how it should inform my actions from here. Do I have to initialize NSS somewhere else? Should I have already initialized it before getting to this point? Adding a few more words about the "why" here could add a lot more clarity for the reader.